4-HTTP/client/utils/localstorage.js

```diff
@@ -0,0 +1,26 @@
+export class LocalStorage {
+  constructor() {
+    if (!localStorage.getItem("todoList"))
+      localStorage.setItem("todoList", JSON.stringify([]));
+  }
+  getList() {
+    return JSON.parse(localStorage.getItem("todoList"));
+  }
+  #saveList(list) {
+    localStorage.setItem("todoList", JSON.stringify(list));
+  }
+  addTodo(item) {
+    this.#saveList([...this.getList(), item]);
+  }
+  editTodo(newTodo, id) {
+    const newList = this.getList().map((todo) => {
```

Comment:
you don't need to map over all the list.
```suggestion
    const oldList = this.getList();
    const newList = [
    	...oldList.slice(0, newTodoIndex),
    	newTodo,
    	...oldList.slice(newTodoIndex + 1),
    ];
    	
```

Reply:
Got it, since we moved on to use server instead of the Localstorage I don't use this file anymore but thanks for the tip, I will use it for sure in the future.